pyseir/models/seir_model_age.py

```diff
@@ -0,0 +1,786 @@
+                     + self.results['A'] + self.results['I']
+                     + self.results['R'] + self.results['D']
+                     + self.results['HGen'] + self.results['HICU'], label='Total')
+
+            plt.xlabel('Time [days]', fontsize=12)
+            plt.yscale(y_scale)
+            # plt.ylim(1, plt.ylim(1))
+            plt.grid(True, which='both', alpha=.35)
+            plt.legend(framealpha=.5)
+            if xlim:
+                plt.xlim(*xlim)
+            else:
+                plt.xlim(0, self.t_list.max())
+            plt.ylim(1, self.N.sum(axis=0) * 1.1)
+
+            plt.subplot(222)
+
+            plt.plot(self.t_list, self.results['D'], alpha=.4, c='k', lw=1, label='Direct Deaths',
+                     linestyle='-')
+
+            plt.plot(self.t_list, self.results['HGen'], alpha=1, lw=2, c='steelblue',
+                     label='General Beds Required', linestyle='-')
+            plt.hlines(self.beds_general, self.t_list[0], self.t_list[-1], 'steelblue', alpha=1, lw=2, label='ICU Bed Capacity', linestyle='--')
+
+            plt.plot(self.t_list, self.results['HICU'], alpha=1, lw=2, c='firebrick', label='ICU Beds Required', linestyle='-')
+            plt.hlines(self.beds_ICU, self.t_list[0], self.t_list[-1], 'firebrick', alpha=1, lw=2, label='General Bed Capacity', linestyle='--')
+
+            plt.plot(self.t_list, self.results['HVent'], alpha=1, lw=2, c='seagreen', label='Ventilators Required', linestyle='-')
+            plt.hlines(self.ventilators, self.t_list[0], self.t_list[-1], 'seagreen', alpha=1, lw=2, label='Ventilator Capacity', linestyle='--')
+
+            plt.xlabel('Time [days]', fontsize=12)
+            plt.ylabel('')
+            plt.yscale(y_scale)
+            plt.ylim(1, plt.ylim()[1])
+            plt.grid(True, which='both', alpha=.35)
+            plt.legend(framealpha=.5)
+            if xlim:
+                plt.xlim(*xlim)
+            else:
+                plt.xlim(0, self.t_list.max())
+
+            plt.subplot(223)
+            plt.plot(self.t_list, [self.suppression_policy(t) for t in self.t_list], c='steelblue')
+            plt.ylabel('Contact Rate Reduction')
+            plt.xlabel('Time [days]', fontsize=12)
+            plt.grid(True, which='both')
+
+            # Reproduction number through time
+            plt.subplot(224)
+            plt.plot(self.t_list, self.results['Rt'], c='steelblue')
+            plt.ylabel('R(t)')
+            plt.xlabel('Time [days]', fontsize=12)
+            plt.grid(True, which='both')
+
+            plt.tight_layout()
+
+        else:
+            # Plot the data by age group
+            fig, axes = plt.subplots(len(self.age_groups), 2, figsize=(10, 50))
+            for n, age_group in enumerate(self.age_groups):
+                ax1, ax2 = axes[n]
+                ax1.plot(self.t_list, self.results['by_age']['S'][n, :], alpha=1, lw=2, label='Susceptible')
```

Comment:
lots of places where you could use e.g. `for ax, alpha in [(ax1, 0.5), (ax2, 1)]`